tests/e2e_test/kyma_deletion_test.go

```diff
@@ -119,3 +123,93 @@ var _ = Describe("KCP Kyma CR should be deleted successfully when SKR cluster ge
 				Should(Succeed())
 		})
 	})
+
+var _ = Describe("When KCP Kyma CR deleted, Kyma Status Metric should also be deleted",
+	Ordered, func() {
+		channel := "regular"
+		kyma := testutils.NewKymaForE2E("kyma-sample", "kcp-system", channel)
+		remoteNamespace := "kyma-system"
+		GinkgoWriter.Printf("kyma before create %v\n", kyma)
+
+		BeforeAll(func() {
+			// make sure we can list Kymas to ensure CRDs have been installed
+			err := controlPlaneClient.List(ctx, &v1beta2.KymaList{})
+			Expect(meta.IsNoMatchError(err)).To(BeFalse())
+		})
+
+		It("Should create empty Kyma CR", func() {
+			Eventually(CreateKymaSecret, timeout, interval).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient).
+				Should(Succeed())
+			Eventually(controlPlaneClient.Create, timeout, interval).
+				WithContext(ctx).
+				WithArguments(kyma).
+				Should(Succeed())
+			By("verifying kyma is ready")
+			Eventually(CheckKymaIsInState, readyTimeout, interval).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+				Should(Succeed())
+			By("verifying remote kyma is ready")
+			Eventually(CheckRemoteKymaCR, readyTimeout, interval).
+				WithContext(ctx).
+				WithArguments(remoteNamespace, []v1beta2.Module{}, runtimeClient, v1beta2.StateReady).
+				Should(Succeed())
+		})
+
+		It("Kyma reconciliation should remove metric when Kyma CR deleted ", func() {
+			By("getting the current metric count")
+			metricCountBeforeDeletion, err := getKymaStateMetricCount()
+			Expect(err).Should(Not(HaveOccurred()))
+			GinkgoWriter.Printf("Kyma State Metric count before CR deletion: %d", metricCountBeforeDeletion)
+
+			By("deleting KCP Kyma")
+			Eventually(controlPlaneClient.Delete, readyTimeout, interval).
+				WithContext(ctx).
+				WithArguments(kyma).
+				Should(Succeed())
+
+			By("waiting for Kyma CR to be removed")
+			Eventually(CheckKCPKymaCRDeleted, 2*time.Minute, interval).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient).
+				Should(Succeed())
+
+			By("should decrease the metric count")
+			metricCountAfterDeletion, err := getKymaStateMetricCount()
+			Expect(err).Should(Not(HaveOccurred()))
+			GinkgoWriter.Printf("Kyma State Metric count after CR deletion: %d", metricCountAfterDeletion)
+			Expect(metricCountAfterDeletion).To(Equal(metricCountBeforeDeletion - 1))
+		})
+	})
+
+func getKymaStateMetricCount() (int, error) {
```

Comment:
It's better have this function to get each state count, basically this test should verify:
1. before kyma CR removed, the `Ready` state equals 1, 
2. after kyma CR removed, the `Deleting` state should not exists, equals 0

Then this function can be useful for other tests, to have a summary count is too generic.

Reply:
good thinking